tools/xtce_generation/telemetry_generator.py

```diff
@@ -26,298 +64,280 @@ class TelemetryGenerator:
 
-        # Add RestrictionCriteria element to use the given APID for comparison
-        restriction_criteria = Et.SubElement(base_container, "xtce:RestrictionCriteria")
-        comparison = Et.SubElement(restriction_criteria, "xtce:Comparison")
-        comparison.attrib["parameterRef"] = "PKT_APID"
-        comparison.attrib["value"] = f"{self.apid}"
-        comparison.attrib["useCalibratedValue"] = "false"
+            # Add the parameter to the ParameterSet
+            parameter = Et.SubElement(self._parameter_set, "xtce:Parameter")
+            parameter.attrib["name"] = name
+            parameter.attrib["parameterTypeRef"] = name
 
-        # Populate EntryList for packet SequenceContainer
-        packet_entry_list = Et.SubElement(science_container, "xtce:EntryList")
-        parameter_refs = self.pkt.loc[7:, "mnemonic"].tolist()
+            description = Et.SubElement(parameter, "xtce:LongDescription")
+            description.text = parameter_data["description"]
 
-        for parameter_ref in parameter_refs:
-            parameter_ref_entry = Et.SubElement(
-                packet_entry_list, "xtce:ParameterRefEntry"
+            # Add the typeref to the parameter type set
+            parameter_type = Et.SubElement(
+                self._parameter_type_set, "xtce:IntegerParameterType"
             )
-            parameter_ref_entry.attrib["parameterRef"] = parameter_ref
+            parameter_type.attrib["name"] = name
+            parameter_type.attrib["signed"] = "false"
 
-        return telemetry_metadata
+            encoding = Et.SubElement(parameter_type, "xtce:IntegerDataEncoding")
+            encoding.attrib["sizeInBits"] = str(parameter_data["lengthInBits"])
+            encoding.attrib["encoding"] = "unsigned"
 
-    def create_remaining_parameters(self, parameter_set):
+    def _add_parameter_type(self, row):
         """
-        Create XML elements for parameters based on DataFrame rows starting
-        from SHCOARSE (also known as MET).
+        Create parameter types based on 'dataType' for the unique 'lengthInBits' values.
+        This will loop through the unique lengths and create a ParameterType element
+        for each length representing a data type.
 
         Parameters:
-        - parameter_set: The ParameterSet element where parameters will be added.
+        - parameter_type_set: The ParameterTypeSet element where parameter types are.
+        - unique_lengths: Unique values from the 'lengthInBits' column.
 
-        Returns:
-        - parameter_set: The updated ParameterSet element.
         """
-        # Process rows from SHCOARSE until the last available row in the DataFrame
-        for index, row in self.pkt.iterrows():
-            if index < 7:
-                # Skip rows until SHCOARSE(also known as MET) which are
-                # not part of CCSDS header
-                continue
+        name = f"{row["packetName"]}_{row["mnemonic"]}"
```

Comment:
With this, we are going to have many `xtce:IntegerParameterType` in`xtce:ParameterTypeSet` if we are using this naming convention. is that what you intended? It seems like we are going to have one for each `mnemonic`.

Reply:
Yes, I think we need this because they have to be unique if the same mnemonic has different size or conversion across different packets. (SPARE for instance where it may be 2 bits in one packet and 1 bit in another, ...)

Open to suggestions if there is a better way to handle this.